src/main/scala/com/amazon/deequ/analyzers/Analyzer.scala

```diff
@@ -299,6 +293,56 @@ abstract class GroupingAnalyzer[S <: State[_], +M <: Metric[_]] extends Analyzer
   }
 }
 
+/**
+ * Data Synchronization Analyzer
+ *
+ * @param dfToCompare    DataFrame to compare
+ * @param columnMappings columns mappings
```

Comment:
nit: We can add more documentation here. Are these the key column mappings? What about the other parameter, comparison column mappings? Will that be added in a future PR?

Reply:
thanks, will update documentation. 
I was planning to do this in phased manner instead of one big PR, will definitely do a follow up PR to add that as well.